code/modules/spells/spell_types/lichdom.dm

```diff
@@ -34,7 +34,7 @@
 		for(var/obj/item in hand_items)
 			// I ensouled the nuke disk once. But it's probably a really
 			// mean tactic, so probably should discourage it.
-			if((item.flags_1 & ABSTRACT_1) || (item.flags_1 & NODROP_1) || (item.flags_2 & STATIONLOVING_2))
+			if((item.flags_1 & ABSTRACT_1) || (item.flags_1 & NODROP_1) || item.GetComponent(/datum/component/stationloving))
```

Comment:
Send a signal like `COMSIG_ITEM_IMBUE_SOUL` and make a return value able to cancel it